EasyCommands/CommandParsers/ParsingEngine.cs

```diff
@@ -40,30 +40,16 @@ partial class Program {
                     implicitMainOffset--;
                 }
 
-                var functionIndices = NewList<int>();
-                for (int i = commandStrings.Count - 1; i >= 0; i--) {
-                    if (commandStrings[i].StartsWith(":")) { functionIndices.Add(i); }
-                }
-
+                var functionIndices = Range(0, commandStrings.Count).Where(i => commandStrings[i].StartsWith(":")).Reverse();
                 foreach (int i in functionIndices) {
-                    String functionString = commandStrings[i].Remove(0, 1).Trim();
-                    List<Token> nameAndParams = Tokenize(functionString);
-                    String functionName = nameAndParams[0].original;
-                    nameAndParams.RemoveAt(0);
-                    FunctionDefinition definition = new FunctionDefinition(functionName, nameAndParams.Select(t => t.original).ToList());
-                    functions[functionName] = definition;
-                }
+                    var nameAndParams = Tokenize(commandStrings[i].Remove(0, 1).Trim());
+                    var functionName = nameAndParams[0].original;
 
-                foreach (int i in functionIndices) {
-                    int startingLineNumber = i + 1 + implicitMainOffset;
-                    String functionString = commandStrings[i].Remove(0, 1).Trim();
-                    List<Token> nameAndParams = Tokenize(functionString);
-                    String functionName = nameAndParams[0].original;
+                    functions[functionName] = new FunctionDefinition(functionName, nameAndParams.Skip(1).Select(t => t.original).ToList());
 
```

Comment:
lol glad to have removed this.  i think it was a relic from before dynamic function parsing, where i needed to know the function names up front and so had to do 2 passes.  Glad to simplify this.